From 9e05ac76ba8c648af7d4ca6eefbc5be50a9f427a Mon Sep 17 00:00:00 2001
From: Yongjun Zhang <yzhang@cloudera.com>
Date: Fri, 19 Feb 2016 09:41:22 -0800
Subject: [PATCH 1305/2748] HADOOP-12828. Print user when services are
 started. (Wei-Chiu Chuang via Yongjun Zhang)

(cherry picked from commit a963baba10648859ad88d4ed9a4325b66b1af0d5)
(cherry picked from commit 62ba1e415a34d1b413a67f421d8c28a1aee13e46)

Change-Id: Ia2dd26e756412946289ec682d54e307af429fc7e
---
 .../java/org/apache/hadoop/util/StringUtils.java   |    1 +
 .../hadoop/crypto/key/kms/server/KMSWebApp.java    |    1 +
 2 files changed, 2 insertions(+)

diff --git a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/StringUtils.java b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/StringUtils.java
index 9625845..017eb8c 100644
--- a/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/StringUtils.java
+++ b/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/StringUtils.java
@@ -649,6 +649,7 @@ static void startupShutdownMessage(Class<?> clazz, String[] args,
     LOG.info(
         toStartupShutdownString("STARTUP_MSG: ", new String[] {
             "Starting " + classname,
+            "  user = " + System.getProperty("user.name"),
             "  host = " + hostname,
             "  args = " + Arrays.asList(args),
             "  version = " + VersionInfo.getVersion(),
diff --git a/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMSWebApp.java b/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMSWebApp.java
index 194ecd6..1474463 100644
--- a/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMSWebApp.java
+++ b/hadoop-common-project/hadoop-kms/src/main/java/org/apache/hadoop/crypto/key/kms/server/KMSWebApp.java
@@ -124,6 +124,7 @@ public void contextInitialized(ServletContextEvent sce) {
       LOG.info("-------------------------------------------------------------");
       LOG.info("  Java runtime version : {}", System.getProperty(
           "java.runtime.version"));
+      LOG.info("  User: {}", System.getProperty("user.name"));
       LOG.info("  KMS Hadoop Version: " + VersionInfo.getVersion());
       LOG.info("-------------------------------------------------------------");
 
-- 
1.7.9.5

